src/pages/dar_collection_review/ApplicationInformation.js

```diff
@@ -53,6 +61,28 @@ const generateLabelSpanContents = (labelValue, key,  spanValue, isLoading) => {
   );
 };
 
+// function to generate application details content
+const dynamicRowGeneration = (rowElementMaxCount, appDetailLabels, loading) => {
+  const labelArray = appDetailLabels.map(label => {
```

Comment:
With `map`, not returning a value doesn't skip over the value, it just assigns undefined for that element in the array, which will have a negative impact on the chunking method that occurs later on. 

What you should do is use lodash's [filter()](https://lodash.com/docs/4.17.15#filter) with `appDetailLabels` so that only the elements with a valid `label[0]` value (non-empty string, non-empty object, non-empty array, booleans) show up. The filter method will return a new array of the appDetailLabels that can be processed with `generateLabelSpanContents`

You have the logic checks set up in your `map` (minus the adjustments stated in the later comments), you just need to use it before this map function to filter for the valid values.